File: BasicAppSample/app/src/javaVersion/java/com/example/basicappsample/MainActivity.java
package com.example.basicappsample;

import android.os.Bundle;
import android.view.KeyEvent;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.basicappsample.databinding.ActivityMainBinding;
import com.example.basicappsample.helpers.AnimationHelper;
import com.example.basicappsample.helpers.KeyboardHelpable;

public class MainActivity extends AppCompatActivity implements KeyboardHelpable {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.textHello.setText("Hello, Android Java");
        AnimationHelper.startAlphaAnimation(binding.textHello);
        binding.editEmail.setOnKeyListener((v, keyCode, event) -> {
            if (event.getAction() == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_ENTER) {
                hideKeyboard(MainActivity.this);
                return true;
            } else {
                return false;
            }
        });
    }
}

File: BasicAppSample/app/src/kotlinVersion/java/com/example/basicappsample/MainActivity.kt
package com.example.basicappsample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.KeyEvent
import com.example.basicappsample.databinding.ActivityMainBinding
import com.example.basicappsample.helpers.AnimationHelper
import com.example.basicappsample.helpers.KeyboardHelpable

class MainActivity : AppCompatActivity(), KeyboardHelpable {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater).also {
            setContentView(it.root)
        }
        binding.textHello.text = "Hello, Android Kotlin"
        AnimationHelper.startAlphaAnimation(binding.textHello)

        binding.editEmail.setOnKeyListener { _, keyCode, event ->
            if (event.action == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_ENTER) {
                hideKeyboard()
                return@setOnKeyListener true
            } else {
                return@setOnKeyListener false
            }
        }
    }
}